src/cmdstan/stansummary.cpp

```diff
@@ -26,503 +26,181 @@ int main(int argc, const char *argv[]) {
-      vm);
-  po::notify(vm);
-
+  boost::program_options::variables_map vm;
+  try {
+    boost::program_options::store(
+        boost::program_options::command_line_parser(argc, argv)
+            .options(desc)
+            .positional(p)
+            .run(),
+        vm);
+    boost::program_options::notify(vm);
+  } catch (const boost::program_options::error &e) {
+    std::cout << "Invalid argument: " << e.what();
+    return -1;
+  }
   if (vm.count("help")) {
     std::cout << desc << std::endl;
     return 0;
   }
 
-  if (vm.count("sig_figs")) {
-    std::cout << "sig_figs " << vm["sig_figs"].as<int>();
-    if (vm["sig_figs"].defaulted())
-      std::cout << " (default)";
-    std::cout << std::endl;
-  }
-  if (vm.count("autocorr"))
-    std::cout << autocorr_idx << std::endl;
-  if (vm.count("csv_filename"))
-    std::cout << "csv_filename " << csv_filename << std::endl;
-  if (vm.count("percentiles")) {
-    std::cout << "percentiles " << percentiles_spec;
-    if (vm["sig_figs"].defaulted())
-      std::cout << " (default)";
-    std::cout << std::endl;
-  }
+  // Validate command line args
   if (vm.count("input_files")) {
+    for (size_t i = 0; i < filenames.size(); ++i) {
+      if (FILE *file = fopen(filenames[i].c_str(), "r")) {
+        fclose(file);
+      } else {
+        std::cout << "Invalid input file: " << filenames[i] << ", exiting."
+                  << std::endl;
+        return -1;
+      }
+    }
     if (filenames.size() == 1)
-      std::cout << "input_file ";
+      std::cout << "Input file: ";
     else
-      std::cout << "input_files ";
+      std::cout << "Input files: ";
     size_t i = 0;
-    for (std::vector<std::string>::iterator it = filenames.begin();
-         it != filenames.end(); it++, ++i) {
-      std::cout << (*it);
+    for (size_t i = 0; i < filenames.size(); ++i) {
+      std::cout << filenames[i];
       if (i < filenames.size() - 1)
         std::cout << ", ";
     }
     std::cout << std::endl;
+  } else {
+    std::cout
+        << "No Stan csv file(s) specified, expecting one or more filenames."
+        << std::endl;
+    std::cout << desc << std::endl;
+    return -1;
+  }
+  if (vm.count("csv_filename")) {
+    if (FILE *file = fopen(csv_filename.c_str(), "w")) {
+      fclose(file);
+    } else {
+      std::cout << "Invalid output csv file: " << csv_filename << ", exiting."
+                << std::endl;
+      return -1;
+    }
+    std::cout << "csv_filename " << csv_filename << std::endl;
+  }
+  if (vm.count("sig_figs") && !vm["sig_figs"].defaulted()) {
+    if (sig_figs < 0 || sig_figs > 100) {
+      std::cout << "Bad value for option --sig_figs: "
+                << vm["sig_figs"].as<int>() << ", exiting." << std::endl;
+      std::cout << desc << std::endl;
+      return -1;
+    }
+    std::cout << "sig_figs " << vm["sig_figs"].as<int>() << std::endl;
+  }
+  if (vm.count("autocorr")) {
+    if (autocorr_idx < 1 || autocorr_idx > filenames.size()) {
```

Comment:
`autocorr_idx` is the index for the chain.
you're thinking of the lag, which is determined inside the `autocorrelation` function - last function in `stansummary_helper.hpp`.  lines 598-605.   if you have 100-999 draws, lag is 2, 1000 -9999 draws the lag is 3, 10K draws the lag is 4, etc.

this is legacy code, it could be cleaned up, and made a whole lot more general and generally useful.   that's for another issue/PR